ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoice.java

```diff
@@ -150,6 +155,7 @@ public ForkChoice(
       final BlobSidecarManager blobSidecarManager,
       final ForkChoiceNotifier forkChoiceNotifier,
       final MergeTransitionBlockValidator transitionBlockValidator,
```

Comment:
We likely do not need to have a dumper here, as this is the method used in multiple tests. My suggestion would be to have a `NoOpDebugDataDumper` that does nothing, and can be passed to `this(..)`.

This way, you only need to pass a "real" dumper on the constructor that is used in the production code, when we initialize ForkChoice.

(another benefit is that it won't require you to change that many tests adding the mocks)